libs/libarchfpga/src/read_fpga_interchange_arch.cpp

```diff
@@ -2243,6 +2327,29 @@ struct ArchReader {
         }
     }
 
+    void add_segment_with_default_values(t_segment_inf& seg, std::string name) {
+        // Use default values as we will populate rr_graph with correct values
+        // This segments are just declaration of future use
+        seg.name = name;
+        seg.length = 1;
+        seg.frequency = 1;
+        seg.Rmetal = 1e-12;
+        seg.Cmetal = 1e-12;
+        seg.parallel_axis = BOTH_AXIS;
+
+        // TODO: Only bi-directional segments are created, but it the interchange format
+        //       has directionality information on PIPs, which may be used to infer the
+        //       segments' directonality.
+        seg.directionality = BI_DIRECTIONAL;
+        seg.arch_wire_switch = 1;
+        seg.arch_opin_switch = 1;
+        seg.cb.resize(1);
+        seg.cb[0] = true;
+        seg.sb.resize(2);
+        seg.sb[0] = true;
+        seg.sb[1] = true;
+    }
+
```

Comment:
This can actually be a static function outside of the ArchReader class. It could be put on top with the other static functions

Reply:
Fixed in next commit